vignettes/sass.Rmd

```diff
@@ -1,165 +1,389 @@
-
-It is best practice to define your [variables](http://sass-lang.com/guide#topic-2) separately, and then [import](http://sass-lang.com/guide#topic-5) them into the main Sass file. For example, we'll define a color variable and use it to set the body's background color.
+[Sass](https://sass-lang.com/) is the most widely used, feature rich, and stable CSS extension language available. It has become an essential tool in modern web development because of its ability to _reduce complexity_ and _increase composability_ when styling a website. For a basic example, suppose want to use the same color for multiple styling rules in your website (e.g., hyperlinks and buttons). With CSS, you'd have to repeat that color every time it is used in styling rule, so in a large project, changing at a later point can be tedious and error-prone.
 
 ```css
-/* _variables.scss */
-$color: #FFFFFF;
+a { 
+  color: salmon; 
+}
+button {
+  background-color: salmon;
+}
 ```
+
+With Sass, you could store this color in a Sass variable and use it in the styling rules to produce the same CSS, resulting in a single entry point for the color's value. This simple Sass tool can make CSS styling a lot easier to reason about, making styles a lot easier to customize and maintain.
+
 ```css
-/* new-style.scss */
-@import 'variables';
-body {
-  background-color: $color;
+$main-color: salmon;
+a { 
+  color: $main-color; 
+}
+button {
+  background-color: $main-color;
 }
 ```
 
-The following [shiny app](https://gallery.shinyapps.io/sass-color) will demonstrate this example. We'll use Dean Attali's [colourpicker](https://github.com/daattali/colourpicker) to choose a color and write it to `_variables.scss`.
+Sass variables are not the only Sass tool useful for reducing complexity. For Sass and **sass** newcomers, this vignette first covers how to use basic Sass tools like [variables](#variables), [mixins](#mixins), and [functions](#functions) in the **sass** R package. After the basics, you'll also learn how to write [composable **sass**](#layers) that allows users to easily override your styling defaults and developers to [include](#includes) your Sass in their styling projects. You'll also learn how to control the CSS output (e.g., [compress](#compression) and [cache](#caching) it), and how to use it in [**shiny**](#shiny) & [**rmarkdown**](#rmarkdown). 
+
+By mastering these concepts, you'll not only be able to leverage the advantages of using Sass over CSS, but you'll also have the basis needed to develop R interfaces to Sass projects that allow users to easily customize your styling templates _without any knowledge of Sass/CSS_. For an example, see the [**bootstraplib** R package](https://github.com/rstudio/bootstraplib), which provides a interface to [Bootstrap Sass](https://getbootstrap.com/docs/4.0/getting-started/theming/) through easy-to-use functions like `bs_theme_add_variables()`.
 
```

Comment:
It might be nice to have a section before you start in on "Sass Basics" to talk about the `sass()` function:

1. That it takes Sass as input, and returns CSS as output
2. That the input can be a Sass string, a filename (using `sass_file`), or one of several kinds of R objects
3. (Maybe) that you don't need to install sass separately, everything is included in the R package

Reply:
Yea, the README sorta covers that, but I suppose it won't hurt to touch on it again briefly